### PR TITLE
App Configuration Merging 2.0.0 libraries

### PR DESCRIPTION
**Pull request by mrm9084**
Brings azure-spring-cloud-appconfiguration-config into master with spring-cloud-azure-appconfiguration-config

### CODE REVIEW

File: eng/versioning/version_client.txt [hunk 1]
```diff
@@ -124,6 +124,11 @@ com.microsoft.azure:spring-cloud-azure-appconfiguration-config;1.3.0;1.4.0-beta.
 com.microsoft.azure:spring-cloud-azure-feature-management-web;1.3.0;1.4.0-beta.1
 com.microsoft.azure:spring-cloud-azure-feature-management;1.3.0;1.4.0-beta.1
 com.microsoft.azure:spring-cloud-starter-azure-appconfiguration-config;1.3.0;1.4.0-beta.1
+com.azure.spring:azure-spring-cloud-appconfiguration-config-web;2.0.0-beta.1
```
Comment (alzimmermsft) [inline eng/versioning/version_client.txt:127]: Formatting is slightly off here, it is missing the current version.

`<artifact identifier>;<last release GA, or beta if no GA>;<current development version>`

So, this should be:

```suggestion
com.azure.spring:azure-spring-cloud-appconfiguration-config-web;2.0.0-beta.1;2.0.0-beta.1
```

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/pom.xml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      <version>2.4.5</version> <!-- {x-version-update;org.springframework.boot:spring-boot-starter-actuator;external_dependency} -->
+      <optional>true</optional>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.cloud</groupId>
+      <artifactId>spring-cloud-bus</artifactId>
+      <version>3.0.2</version> <!-- {x-version-update;org.springframework.cloud:spring-cloud-bus;external_dependency} -->
+      <optional>true</optional>
+      <scope>compile</scope>
```
Comment (saragluna) [inline sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/pom.xml:42]: The default scope is `compile`, so we could omit this.

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/pom.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-web</artifactId>
+      <version>2.4.5</version> <!-- {x-version-update;org.springframework.boot:spring-boot-starter-web;external_dependency} -->
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-actuator</artifactId>
+      <version>2.4.5</version> <!-- {x-version-update;org.springframework.boot:spring-boot-starter-actuator;external_dependency} -->
+      <optional>true</optional>
+      <scope>compile</scope>
```
Comment (saragluna) [inline sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/pom.xml:35]: The default scope is `compile`, so we could omit this.

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/pom.xml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>3.6.28</version><!-- {x-version-update;org.mockito:mockito-core;external_dependency} -->
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
```
Comment (saragluna) [inline sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/pom.xml:57]: Spring Boot 2.4.x has removed the vintage engine dependency by default, https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.4-Release-Notes#junit-5s-vintage-engine-removed-from-spring-boot-starter-test, will this affect our test cases?

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/resources/META-INF/spring.factories [hunk 5]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
+com.microsoft.azure.spring.cloud.config.web.AppConfigurationWebAutoConfiguration
```
Comment (saragluna) [inline sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/resources/META-INF/spring.factories:2]: If we follow the track2 Spring guidelines, all Spring related packages should start with `com.azure.spring.`

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/pom.xml (hunk 4 above)
Comment (mrm9084) [reply sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/pom.xml:57]: @saragluna I have been testing on 2.4 for a while now and haven't seen any issues.

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/resources/META-INF/spring.factories (hunk 5 above)
Comment (mrm9084) [reply sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/resources/META-INF/spring.factories:2]: @saragluna I will add it to my list and update in the next preview version.

**Review by saragluna — APPROVED** [review]
LGTM

**Pull request merged**